astacus/node/download.py

```diff
@@ -117,6 +124,29 @@ def _cb(*, map_in: Sequence[ipc.SnapshotFile], map_out: Sequence[ipc.SnapshotFil
             with contextlib.suppress(FileNotFoundError):
                 absolute_path.unlink()
 
+        if self.copy_dst_owner:
+            # Adjust owner of created files and folders to be like the owner of dst
+            dst_owner_uid = self.dst.stat().st_uid
+            chowned_paths: set[str] = set()
+            for snapshotfile in snapshotstate.files:
+                absolute_path = self.dst / snapshotfile.relative_path
+                chown_candidate = absolute_path
+                while chown_candidate != self.dst and chown_candidate.stat().st_uid != dst_owner_uid:
+                    chowned_paths.add(str(chown_candidate.relative_to(self.dst)))
+                    chown_candidate = chown_candidate.parent
+            if chowned_paths:
+                astacus_user = getpass.getuser()
+                dst_owner_gid = self.dst.stat().st_gid
+                # We're very specific to allow a sufficiently restrictive sudoers configuration
```

Comment:
Can we comment to link to where we set the sudoers? I do wonder if managing this just by adding the two users to the same group that we permission to the files is easier to follow, but as is I think this makes sense to me. We could also just explicitly pass the user in instead of unmasking the folder, since we know when instantiating the class.

Reply:
We also share a group between astacus and the service that we backup, but that's not sufficient to avoid all permission errors when doing operations on the restored file (for instance, ClickHouse uses utime on some files and folders, which require being the exact owner, not merely having write access, hardlink operations are also quite restricted).

Where we set the sudoers is dependent on how astacus is deployed, aiven does it one way in core but it's not generalizable. (Or you just mean describing the typical content of a reasonable file in `/etc/sudoers.d/something` ?)

Reply:
I was hoping the former was standardized, but otherwise I guess the latter?

Reply:
No point relishing just for that though